### ISSUE TITLE
del not working

### ISSUE THREAD

**Issue by Rc85** [redis/node-redis#2267]
Trying to delete a single key and array of keys. It shows delete count as 1, but when going into `redis-cli` and executing `keys *`, the key is still there.

**Environment:**
 - **Node.js Version**:  16.15.0
 - **Redis Server Version**: 3.0.504
 - **Node Redis Version**: 4.3.1
 - **Platform**: Windows 10
![image](https://user-images.githubusercontent.com/22307591/190847451-8e879ff6-65e4-40e0-828b-ccbcd8c11d06.png)

Top: The code I use to create client, connect, and executing delete.
Bottom left is the result of the `console.log`.
Bottom right my `redis-cli` and `keys *` is executed after `console.log` shows `1`.

**Comment by leibale** [comment]
Are you sure that `SESSION_DB` is `0`?

**Comment by Rc85** [comment]
> Are you sure that `SESSION_DB` is `0`?

Yes. Just tested again by hard coding 0 to `database`. Still shows `1` in `console.log` and record in redis is still there.

**Comment by leibale** [comment]
Can you please try this one:
```javascript
console.log('SET', await client.set('testkey', 'value'));
console.log('GET', await client.get('testkey'));
console.log('DEL', await client.del('testkey'));
console.log('GET', await client.get('testkey'));
```

**Comment by Rc85** [comment]
> Can you please try this one:
> 
> ```js
> console.log('SET', await client.set('testkey', 'value'));
> console.log('GET', await client.get('testkey'));
> console.log('DEL', await client.del('testkey'));
> console.log('GET', await client.get('testkey'));
> ```

That works. I went ahead and did a `SET` in `redis-cli` and then deleting from my code, it works. So I finally narrowed down as to why only keys set by `express-session` is not being deleted. It turns out it was because I had the `resave` option set to `true` and as per the documentation says, it creates a race condition where the session is resaved back to the database. I guess it got deleted and then resaved again.

Thanks for helping me debug this, much appreciate it.
